app/controllers/keywords_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class KeywordsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @keywords = Keyword.all
+    @keyword = Keyword.new
+  end
+
+  def create
+    if keywords_parse_csv
+      flash[:notice] = I18n.t('csv.upload_success')
+    else
+      flash[:alert] = keywords_form.errors.full_messages.first
+    end
+
+    redirect_to keywords_path
+  end
+
+  private
+
+  def keywords_form
+    @keywords_form ||= KeywordsForm.new(current_user)
+  end
+
+  def keywords_parse_csv
+    keywords_form.save(params[:keyword][:keywords_file])
+  end
```

Comment:
- Do the private methods necessary as we just use them once? 💭 
- How about using the [strong param](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters) to align with [our convention](https://nimblehq.co/compass/development/code-conventions/ruby/ruby-on-rails/#convention-30)?

```suggestion
  def create
    keywords_form ||= KeywordsForm.new(current_user)

    if keywords_form.save(keyword_params)
      flash[:notice] = I18n.t('csv.upload_success')
    else
      flash[:alert] = keywords_form.errors.full_messages.first
    end

    redirect_to keywords_path
  end

  private

  def keyword_params
    params.require(:keyword).permit(:keywords_file)
  end
```

Reply:
Yup, I like this implementation 😄 Updated in https://github.com/aetheryna/google-scraper-ruby/pull/34/commits/25b9f583ac59d64e0e10127480a5f768837eeeaa